app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/VatRegisteredUkController.scala

```diff
@@ -99,27 +100,26 @@ class VatRegisteredUkController @Inject() (
           yesNoAnswer =>
             subscriptionDetailsService.cacheVatRegisteredUk(yesNoAnswer).flatMap {
               _ =>
-                if (isInReviewMode)
-                  if (yesNoAnswer.isYes)
-                    Future.successful(Redirect(VatDetailsController.reviewForm(service).url))
-                  else
+                (isInReviewMode, yesNoAnswer.isYes, featureFlags.useNewVATJourney) match {
+                  case (false, true, false) =>
+                    Future.successful(Redirect(VatDetailsControllerOld.createForm(service).url))
+                  case (true, true, false) =>
+                    Future.successful(Redirect(VatDetailsControllerOld.reviewForm(service).url))
+                  case (false, true, true) => Future.successful(Redirect(VatDetailsController.createForm(service).url))
+                  case (true, true, true)  => Future.successful(Redirect(VatDetailsController.reviewForm(service).url))
+                  case (false, false, _) =>
+                    subscriptionDetailsService.clearCachedUkVatDetails flatMap { _ =>
+                      Future.successful(
+                        Redirect(
+                          subscriptionFlowManager.stepInformation(VatDetailsSubscriptionFlowPage).nextPage.url(service)
+                        )
+                      )
+                    }
+                  case (true, false, _) =>
```

Comment:
Since all of these produce a Future.successful(Redirect(...))) it would read more cleanly if you returned a Future.successful(Redirect(target))) where target is the match returning a URL.

Reply:
Love the match though